plugins/scale/wayfire/plugins/scale-signal.hpp

```diff
@@ -0,0 +1,79 @@
+/* definition of filters for scale plugin and activator */
+
+#ifndef SCALE_SIGNALS_H
+#define SCALE_SIGNALS_H
+
+#include <wayfire/object.hpp>
+#include <wayfire/view.hpp>
+#include <vector>
+#include <algorithm>
+
+/**
+ * name: scale-filter
+ * on: output
+ * when: This signal is sent from the scale plugin whenever it is updating the
+ *   list of views to display, with the list of views to be displayed in
+ *   views_shown. Plugins can move views to views_hidden to request them not to
+ *   be displayed by scale.
+ *
+ * Note: it is an error to remove a view from views_shown without adding it to
+ *   views_hidden; this will result in views rendered in wrong locations.
+ *
+ * If multiple plugins are connected to this signal, they are called in the
+ * order defined by the logic in signal_provider_t; plugins should not depend
+ * on being called in a predictable order. Specifically, plugins should not
+ * expect views_hidden to be empty (and should not call clear() on it). It is OK
+ * for a plugin to move a view from views_hidden to views_shown, but this will
+ * likely not have predictable results.
+ *
+ * TODO: should there be a way for plugins to specify which order they want to
```

Comment:
I think this is fine as it is now. I expect all consumers of this API to move from visible to hidden based on some criteria, so it wouldn't make a difference how they are ordered.